api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -38,8 +39,27 @@
    * Add a rewrite that replaces one set of files with another set that contains the same data.
    *
    * @param filesToDelete files that will be replaced (deleted), cannot be null or empty.
-   * @param filesToAdd files that will be added, cannot be null or empty.
+   * @param filesToAdd    files that will be added, cannot be null or empty.
    * @return this for method chaining
    */
-  RewriteFiles rewriteFiles(Set<DataFile> filesToDelete, Set<DataFile> filesToAdd);
+  default RewriteFiles rewriteFiles(Set<DataFile> filesToDelete, Set<DataFile> filesToAdd) {
```

Comment:
How about adding `RewriteFiles rewriteDeletes(Set<DeleteFile> fileToDelete, Set<DeleteFile> filesToAdd` for rewrite deletes?

Reply:
I think you want to expose this API for replacing equality deletions with positional deletions,  for me that seems like an internal usage we may don't have to define such a specific API for end users.  I prefer to expose the following common API to end users,  for our internal usage we could rewrite files based on that one.